stores/pg/src/vector/index.ts

```diff
@@ -215,18 +216,33 @@ export class PgVector extends MastraVector {
         throw new Error('PostgreSQL vector extension is not available. Please install it first.');
       }
 
-      // Try to create extension
-      await client.query('CREATE EXTENSION IF NOT EXISTS vector');
+      // Get advisory lock using hash of index name
+      const hash = createHash('sha256').update(indexName).digest('hex');
+      const lockId = BigInt('0x' + hash.slice(0, 8)) % BigInt(2 ** 31); // Take first 8 chars and convert to number
+      const acquired = await client.query('SELECT pg_try_advisory_lock($1)', [lockId]);
 
-      // Create the table with explicit schema
-      await client.query(`
+      if (!acquired.rows[0].pg_try_advisory_lock) {
+        // Wait for other process to finish
+        await client.query('SELECT pg_advisory_lock($1)', [lockId]);
+      }
```

Comment:
If it was locked doesn't that mean we can return early here? Since the other code execution path would've already created the index

Reply:
That's a good point. let me try adding that

Reply:
probably still want to await it being released and then return early

Reply:
Looks like it works to do that, but it might be worth to confirm the index is created if the path doesn't have the lock. I'll add something and we can see if we like it.